### DISCUSSION TITLE
Box64 compile args to use when running on M1 but inside a VM

### DISCUSSION THREAD

**Discussion by sonroyaalmerol** [ptitSeb/box64#1667] [Q&A] [answered]
Hi! Hopefully this is the right place to ask.

In the compiling instructions, I can see a specific arg for M1 `-D M1=1`. Is this only applicable for M1s with Asahi? Can I also use this if I'm planning to run Box64 within a VM on an M1? If not, what would be the "best" compile args to use?

If it matters, the VM will be running with 4k page size afaik.

Thanks!

**Comment by ptitSeb** [comment, marked as answer]
the `-DM1=1` is used mainly to select CPU optimisation for the build. The Pagesize is read in runtime. So keep `-DM1=1` it will  be the best settings.

**Reply by sonroyaalmerol** [reply]
That's good to hear. Thanks!
